<launch>

    <!-- SYSTEM MODEL -->
    <!-- vehicle parameters -->
    <param name="m"             type="double"   value="2.424" />
    <param name="rho"             type="double"   value="1.225" />    
    <param name="lr"            type="double"   value="0.1203" />
    <param name="lf"            type="double"   value="0.1377" />
    <param name="Cm0"            type="double"   value="9.46853" />
    <param name="Cm1"            type="double"   value="0.667237" />
    <param name="C0"            type="double"   value="2.61049" />
    <param name="C1"            type="double"   value="-0.00213596" />
    <param name="Cd_A"            type="double"   value="-0.466602" />
    <param name="Caf"            type="double"   value="1.3958" />
    <param name="Car"            type="double"   value="1.6775" />
    <param name="Iz"            type="double"   value="0.02" />
    

    <!-- for slew rate
        <param name="max_vel"          type="double"   value="5.0" />
    <param name="min_vel"          type="double"   value="-0.5" />
    <param name="dutycycle_max"     type="double"   value="1.0" />
    <param name="dutycycle_min"     type="double"   value="-1.0" />
    -->

    <param name="max_vel"          type="double"   value="5.0" />
    <param name="min_vel"          type="double"   value="-0.01" />
    <param name="dutycycle_max"     type="double"   value="1.0" />
    <param name="dutycycle_min"     type="double"   value="-0.00" />
    <param name="steer_max"         type="double"   value="0.349" />
    <param name="steer_min"         type="double"   value="-0.349" />
    <param name="lat_e_max"         type="double"   value="0.25" />   <!-- like halfWidth-->
    <param name="orient_e_max"      type="double"   value="0.5" />


    <!-- Track parameters -->
    <!--param name="trackShape" type="str" value="L_shape" /-->
    <param name="halfWidth"     type="double"   value="0.25" />

    <param name="trackShape" type="str" value="oval_iri" />


    <!-- TRAJECTORY PLANNING -->
    <node pkg="planner" type="mpc_planner.py" name="trajectory_planner" output="screen"> 
         <param name="Planning_Track"   type="int"      value="1" /> 
         <param name="N"                type="int"      value="40" /> 
         <param name="Hz"        type="int"      value="25" /> 
         <param name="vel_ref"        type="double"      value="1.2" /> <!-- reference velocity --> 
         <param name="halfWidth"        type="double"   value="0.25" />
         <param name="Visualization"    type="int"      value="0" />
         <param name="Mode"          type="int"      value="1" /> <!-- 1: for testing without estimator, 2: solution w.r.t the estimator data 3: keyboard input with estimator testing  -->
        <param name="integral_action"   type="bool"     value="false"/>
        <param name="soft_constraints"  type="bool"     value="false"/>
    </node>   

</launch>
